ilastik/applets/dataSelection/dataSelectionSerializer.py

```diff
@@ -331,6 +319,33 @@ def updateWorkingDirectory(self, newpath, oldpath):
         self.topLevelOperator.WorkingDirectory.setValue(newdir)
         self._projectFilePath = newdir
 
+    def updateLegacyProjectFile(self, file: File):
+        """
+        Presence of UrlDatasetInfo indicates this is a pre-multiscale project file created in HBP mode.
+        These projects only supported working on the first scale in the list of resolutions
+        of a Precomputed volume.
+        We reverse this list in RESTfulPrecomputedChunkedVolume._init_config, so we need to
+        use the last scale when storing the object as a new MultiscaleUrlDatasetInfo.
+        """
+        try:
+            info_dir = file[self.topGroupName]["infos"]
+        except KeyError:
+            return
+
+        for laneIndex, (_, laneGroup) in enumerate(sorted(info_dir.items())):
+            for roleName, infoGroup in sorted(laneGroup.items()):
+                if "__class__" not in infoGroup:
+                    continue
+                loaded_class_name = infoGroup["__class__"][()].decode("utf-8")
+                if loaded_class_name == "UrlDatasetInfo":
+                    old_nickname = infoGroup["nickname"][()].decode("utf-8")
+                    del infoGroup["__class__"]
+                    del infoGroup["nickname"]
+                    infoGroup["__class__"] = "MultiscaleUrlDatasetInfo".encode("utf-8")
+                    infoGroup["nickname"] = MultiscaleUrlDatasetInfo.nickname_from_url(old_nickname).encode("utf-8")
+                    infoGroup["working_scale"] = "-1".encode("utf-8")
+                    infoGroup["scale_locked"] = "True".encode("utf-8")
```

Comment:
1. `laneIndex` doesn't seem to be used
2. Sorting doesn't seem to be necessary (maybe for debugging?..)
3. `bytes.decode` and `str.encode` do `utf-8` by default
4. Byte literals could be used directly instead of encoding them from strings

```suggestion
        for laneGroup in info_dir.values():
            for infoGroup in laneGroup.values():
                try:
                    if infoGroup["__class__"][()] != b"UrlDatasetInfo":
                        continue
                except KeyError:
                    continue
    
                old_nickname = infoGroup["nickname"][()].decode()
                new_nickname = MultiscaleUrlDatasetInfo.nickname_from_url(old_nickname).encode()

                del infoGroup["__class__"]
                infoGroup["__class__"] = b"MultiscaleUrlDatasetInfo"
                
                del infoGroup["nickname"]
                infoGroup["nickname"] = new_nickname
                
                infoGroup["working_scale"] = b"-1"
                infoGroup["scale_locked"] = b"True"
```